src/home/components/PetWeightPage/petWeightPage.spec.js

```diff
@@ -1,16 +1,188 @@
-import { describe, expect, it } from 'vitest';
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
 import PetWeightPage from './index';
+import UploadImage from '../../../components/UploadImage';
+import RangeSlider from '../../../components/RangeSlider';
+import TextInput from '../../../components/TextInput';
+import RadioButton from '../../../components/RadioButton';
+import Button from '../../../components/Button';
+
+let instance;
+
+vi.mock('../../../components/UploadImage', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/RangeSlider', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    setValue: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/TextInput', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    setValue: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/RadioButton', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    isChecked: vi.fn().mockImplementation(function isChecked() {
+      return this === instance.radioKG;
+    }),
+    getValue: vi.fn().mockImplementation(function getValue() {
+      return this === instance.radioKG ? 'kg' : 'lb';
+    }),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
+
+vi.mock('../../../components/Button', () => ({
+  __esModule: true,
+  default: vi.fn().mockImplementation(() => ({
+    mount: vi.fn(),
+    listen: vi.fn(),
+    selected: {
+      get: vi.fn().mockReturnValue({
+        classList: {
+          add: vi.fn(),
+        },
+      }),
+    },
+  })),
+}));
 
 const propsMock = {
   petPhoto: 'https://via.placeholder.com/150',
 };
 
 describe('PetWeightPage', () => {
+  beforeEach(() => {
+    instance = new PetWeightPage(propsMock);
+  });
```

Comment:
beforeEach() não é uma boa prática em testes, ainda mais que só ta inicializando o componente, você pode fazer isso no começo de cada It, fica mais legível.